pkg/build/controller/controller.go

```diff
@@ -325,7 +333,7 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 		}
 		glog.V(4).Infof("Updating build %s/%s status %s -> %s%s", build.Namespace, build.Name, build.Status.Phase, nextStatus, reason)
 		build.Status.Phase = nextStatus
-		build.Status.Message = ""
```

Comment:
I think Message and Reason should always be updated together, so i suggest moving this up into the individual case statements where Reason is being reset.